cmd/serve.go

```diff
@@ -43,11 +53,80 @@ func (c *serveCmd) Command() *cobra.Command {
 			ConfigKey:      &cfg.WalletSigningKey,
 			Required:       true,
 		},
+		{
+			Name:           "supported-assets",
+			Usage:          `A collection of supported assets (i.e. USDC). This value is an array of JSON objects. Example: [{"code": "USDC", "issuer": "GBBD47IF6LWK7P7MDEVSCWR7DPUWV3NY3DTQEVFL4NAT4AQH3ZLLFLA5"}]`,
+			OptType:        types.String,
+			CustomSetValue: utils.SetConfigOptionAssets,
+			ConfigKey:      &cfg.SupportedAssets,
+			FlagDefault:    `[{"code": "USDC", "issuer": "GBBD47IF6LWK7P7MDEVSCWR7DPUWV3NY3DTQEVFL4NAT4AQH3ZLLFLA5"}, {"code": "ARST", "issuer": "GB7TAYRUZGE6TVT7NHP5SMIZRNQA6PLM423EYISAOAP3MKYIQMVYP2JO"}]`,
+			Required:       true,
+		},
+		{
+			Name:           "distribution-account-private-key",
+			Usage:          "The Distribution Account private key.",
+			OptType:        types.String,
+			CustomSetValue: utils.SetConfigOptionStellarPrivateKey,
+			ConfigKey:      &distributionAccountPrivateKey,
+			Required:       true,
+		},
+		{
+			Name:        "network-passphrase",
+			Usage:       "The Stellar network passphrase",
+			OptType:     types.String,
+			ConfigKey:   &networkPassphrase,
+			FlagDefault: network.TestNetworkPassphrase,
+			Required:    true,
+		},
+		{
+			Name:        "max-sponsored-threshold",
+			Usage:       "The maximum reserves will be sponsored by the distribution account.",
+			OptType:     types.Int,
+			ConfigKey:   &cfg.MaxSponsoredThreshold,
+			FlagDefault: 15,
+			Required:    true,
+		},
+		{
+			Name:        "base-fee",
+			Usage:       "The base fee (in stroops) for submitting a Stellar transaction",
+			OptType:     types.Int,
+			ConfigKey:   &cfg.BaseFee,
+			FlagDefault: 100 * txnbuild.MinBaseFee,
+			Required:    true,
+		},
+		{
+			Name:        "horizon-url",
+			Usage:       "The URL of the Stellar Horizon server where this application will communicate with.",
+			OptType:     types.String,
+			ConfigKey:   &horizonURL,
+			FlagDefault: horizonclient.DefaultTestNetClient.HorizonURL,
+			Required:    true,
+		},
 	}
 	cmd := &cobra.Command{
-		Use:               "serve",
-		Short:             "Run Wallet Backend server",
-		PersistentPreRunE: utils.DefaultPersistentPreRunE(cfgOpts),
+		Use:   "serve",
+		Short: "Run Wallet Backend server",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if err := cfgOpts.RequireE(); err != nil {
+				return fmt.Errorf("requiring values of config options: %w", err)
+			}
+			if err := cfgOpts.SetValues(); err != nil {
+				return fmt.Errorf("setting values of config options: %w", err)
+			}
+
+			signatureClient, err := signing.NewEnvSignatureClient(distributionAccountPrivateKey, networkPassphrase)
+			if err != nil {
+				log.Fatalf("Error instantiating Env signature client: %s", err.Error())
+			}
+
+			cfg.SignatureClient = signatureClient
+			cfg.HorizonClient = &horizonclient.Client{
```

Comment:
I understand that we want to instantiate the signature client here to minimize the exposure of private keys throughout the application, but do you think that we should instantiate other clients here too as opposed to in the `getHandlerDeps` function? Curious to your thoughts.

Reply:
We can instantiate on `getHandlerDeps`.